### DISCUSSION TITLE
2021/11/为Matrix加入推送功能/

### DISCUSSION THREAD

**Discussion by SouthFox-D** [SouthFox-D/SouthFox-D.github.io#55] [博客评论]
https://blog.southfox.me/2021/11/为Matrix加入推送功能/

**Comment by lili202** [comment]
大佬，能不能请你写个bot的搭建教程呢？
感觉 https://github.com/maubot/maubot 这里rss和提醒bot很有用，无奈看不懂啊。

**Reply by SouthFox-D** [reply]
这个程序好像自带了一个管理后台，似乎操作都能通过后台完成……
所以搭建过程可以直接套用之前建站的经验的。

**Comment by yh317** [comment]
你好，请问安卓端要下载什么app，才能看到推送呢？

**Reply by SouthFox-D** [reply]
现在已经不需要在像博客文章那样折腾了，因为 synapse 或是 element 都开始支持 unifiedpush 这个规范 https://unifiedpush.org/ 。

简单点说，只需要安装一个 ntfy https://ntfy.sh/ ，并配置一个实例就可以了。（如果自建 ntfy 实例配置 unifiedpush 可以看相关文档 https://docs.ntfy.sh/config/#example-unifiedpush ）

**Comment by yh317** [comment]
请问unifiedpush墙内能用吗？如果有时间的话可不可以出一期配置教程？我用的docker版synapse，不知道可不可以用？

**Reply by SouthFox-D** [reply]
`unifiedpush` 只是一套规范，只要相应推送应用支持这套规范就可以，相应列表可以参见这个 https://unifiedpush.org/users/distributors/

所以能不能用取决于你的推送应用选择的节点能不能访问（不管公共的或是自建的）。具体的配置教程要看具体要用的推送应用，比如 ntfy 的文档是这个： https://docs.ntfy.sh/config/#example-unifiedpush (如果打算自建的话，如果是官方旗舰节点应该直接就配好了)

---

`synapse` 使用 `unifiedpush` 要看具体客户端（例如 F-Droid 版 element）客户端发现你手机上有相应推送服务例如 ntfy 并且你推送手段选择了使用 ntfy ，那么 element 就会把相关推送地址发回 synapse 注册并在后续推送消息，然后手机上的推送服务得常驻后台监听相关的消息。

所以想接收推送其实都要让相关应用挂在后台，选择自定义推送优势我觉得最大一点是可以在本身就有接收消息（例如站点下线提醒）的需求顺带接收 matrix 的消息。

**Comment by wangzzf** [comment]
大佬ios推送有什么办法吗，不挂魔法，有别的方法吗
